src/mame/jaleco/megasys1.h

```diff
@@ -51,11 +51,13 @@ class megasys1_state : public driver_device
 		m_hardware_type_z = 0;
 	}
 
+	void system_base(machine_config &config);
+
+	void system_B(machine_config &config);
 	void system_B_monkelf(machine_config &config);
+	void system_Bbl(machine_config &config);
 
 	void system_C(machine_config &config);
-	void system_Bbl(machine_config &config);
-	void system_base(machine_config &config);
```

Comment:
The abstract machine configurations (e.g. `system_base`) should be `protected` rather than `public`.  Also may as well put `ATTR_COLD` on them while you’re moving them around.

Reply:
Done. When all MCUs Internal ROMs will be dumped, "iosim" class could be deleted and those abstract methods will not be needed anymore.

I also reordered methods, grouping them in a more clear way.